File: src/main/java/com/mbayou/kick4k/authorization/Scope.java
package com.mbayou.kick4k.authorization;

public enum Scope {
    USER_READ("user:read"),
    CHANNEL_READ("channel:read"),
    CHANNEL_WRITE("channel:write"),
    CHAT_WRITE("chat:write"),
    STREAMKEY_READ("streamkey:read"),
    EVENTS_SUBSCRIBE("events:subscribe"),
    MODERATION_BAN("moderation:ban");

    private final String scope;

    Scope(String scope) {
        this.scope = scope;
    }

    public String getScope() {
        return scope;
    }
}

File: src/main/kotlin/com/mbayou/kick4k/authorization/Scope.kt
package com.mbayou.kick4k.authorization

enum class Scope(val scope: String) {
    USER_READ("user:read"),
    CHANNEL_READ("channel:read"),
    CHANNEL_WRITE("channel:write"),
    CHAT_WRITE("chat:write"),
    STREAMKEY_READ("streamkey:read"),
    EVENTS_SUBSCRIBE("events:subscribe"),
    MODERATION_BAN("moderation:ban"),
}
